Detectors/ITSMFT/ITS/tracking/cuda/include/ITStrackingCUDA/DeviceStoreVertexerGPU.h

```diff
@@ -0,0 +1,262 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \file DeviceStoreVertexer.h
+/// \brief This class serves as memory interface for GPU vertexer. It will access needed data structures from devicestore apis.
+///        routines as static as possible.
+/// \author matteo.concas@cern.ch
+
+#ifndef O2_ITS_TRACKING_DEVICE_STORE_VERTEXER_GPU_H_
+#define O2_ITS_TRACKING_DEVICE_STORE_VERTEXER_GPU_H_
+
+#include "ITStracking/Cluster.h"
+#include "ITStracking/Constants.h"
+#include "ITStracking/Configuration.h"
+#include "ITStracking/Tracklet.h"
+#include "ITStracking/ClusterLines.h"
+#include "ITStrackingCUDA/Array.h"
+#include "ITStrackingCUDA/UniquePointer.h"
+#include "ITStrackingCUDA/Vector.h"
+#include "GPUCommonDef.h"
+
+namespace o2
+{
+namespace its
+{
+namespace GPU
+{
+enum class TrackletingLayerOrder {
+  fromInnermostToMiddleLayer,
+  fromMiddleToOuterLayer
+};
+
+enum class VertexerLayerName {
+  innermostLayer,
+  middleLayer,
+  outerLayer
+};
+
+typedef TrackletingLayerOrder Order;
+class DeviceStoreVertexerGPU final
+{
+ public:
+  DeviceStoreVertexerGPU();
+  ~DeviceStoreVertexerGPU() = default;
+
+  UniquePointer<DeviceStoreVertexerGPU> initialise(const std::array<std::vector<Cluster>, constants::its::LayersNumberVertexer>&,
+                                                   const std::array<std::array<int, constants::index_table::ZBins * constants::index_table::PhiBins + 1>,
+                                                                    constants::its::LayersNumberVertexer>&);
```

Comment:
Why don't you pass this matrix by reference?

Reply:
Sorry, I think I did not understand. Do you mean the IndexTable? Isn't it passed by ref? What am I possibly overlooking?

Reply:
Sorry, I think I did not understand. Do you mean the IndexTable? Isn't it passed by ref? What am I possibly overlooking?

Reply:
ops, sorry, I've miscounted commas, ignore my comment..